internal/service/searchindex/resource_search_index.go

```diff
@@ -396,38 +391,39 @@ func flattenSearchIndexSynonyms(synonyms []admin.SearchSynonymMappingDefinition)
 
 func marshalSearchIndex(fields any) (string, error) {
```

Comment:
unit test?

Reply:
not a strong opinion but don't know if they are value as in other cases because they're local functions that are basically wrappers to json functionality